src/main/java/app/service/impl/OperationSupply.java

```diff
@@ -0,0 +1,21 @@
+package app.service.impl;
+
+import app.FruitStorage;
+import app.model.SupplyFruitBatch;
+import app.service.Operation;
+import java.util.List;
+
+public class OperationSupply implements Operation {
+    @Override
+    public void execute(List<String> data) {
+        SupplyFruitBatch currentFruit = new FruitParserImplementation().parse(data);
```

Comment:
code duplication in all operations. Consider accepting SupplyFruitBatch (or just Fruit) immediately